### ISSUE TITLE
[Funceble] Result of execution 

### ISSUE THREAD

**Issue by funilrys** [StevenBlack/hosts#259]
Hello @StevenBlack ,

As you (may) know, I released and still continue to develop [Funceble](https://github.com/funilrys/funceble).

I wanted to help, so you can find here the result of my last execution against the following list.

As it's marked as **"occasionally"** for the update frequency, 
I also published this message/issue to [AdAway's repository ](https://github.com/AdAway/adaway.github.io/issues/6).

[adaway.github.io@AdAway](https://github.com/funilrys/dead-hosts)  ==> AdAway :x:

Have a nice day/night.

Cheers,

P.S: I plan to test all list of your **"Sources of hosts data unified in this variant"** section.

**Comment by funilrys** [comment]
Hi there,

Just released the results with the following.

As it's marked as **"occasionally"** for the update frequency, 
I also published this message/issue to [AdAway's repository ](https://github.com/AdAway/adaway.github.io/issues/6).

[hosts.extras--add.2o7Net@FadeMind](https://github.com/funilrys/dead-hosts/tree/master/hosts.extras--add.2o7Net%40FadeMind) ==> add.2o7Net :white_check_mark:

No inactive domains found with this list.

Cheers

**Comment by funilrys** [comment]
Hi there,

Just released the results for the following.

As it's marked as **"occasionally"** for the update frequency, 
I also published this message/issue to [FadeMind repository ](https://github.com/FadeMind/hosts.extras/issues/3).

[hosts.extras--add.Dead@FadeMind](https://github.com/funilrys/dead-hosts/tree/master/hosts.extras--add.Dead%40FadeMind) ==> add.Dead :x:

Cheers

**Comment by funilrys** [comment]
Hi there,

Just released the results for the following.

As it's marked as **"occasionally"** for the update frequency, 
I also published this message/issue to [FadeMind's repository ](https://github.com/FadeMind/hosts.extras/issues/3).

[hosts.extras--add.Risk@FadeMind](https://github.com/funilrys/dead-hosts/tree/master/hosts.extras--add.Dead%40FadeMind) ==> add.Risk :x:

Cheers

**Comment by funilrys** [comment]
Hi there,

Just released the results for the following.

As it's marked as **"occasionally"** for the update frequency, 
I also published this message/issue to [FadeMind's repository ](https://github.com/FadeMind/hosts.extras/issues/3).

[hosts.extras--add.Spam@FadeMind](https://github.com/funilrys/dead-hosts/tree/master/hosts.extras--add.Spam%40FadeMind) ==> add.Spam :x:

Cheers

**Comment by funilrys** [comment]
Hi there,

Earlier today I released the following.

As it's marked as **"occasionally"** for the update frequency, 
I also published this message/issue to [FadeMind's repository ](https://github.com/FadeMind/hosts.extras/issues/3).

[hosts.extras--add.Spam@FadeMind](https://github.com/funilrys/dead-hosts/tree/master/hosts.extras--add.Spam%40FadeMind) ==> add.Spam :x:

Cheers

[Badd-Boyz-Hosts@mitchellkrogza](https://github.com/funilrys/dead-hosts) ==> Badd-Boyz-Hosts :x:

**Comment by StevenBlack** [comment]
Hi Nissar @funilrys thanks for this.

Help me understand, can you summarize what does Funceble does?  Are you suggesting any action is required here?  Thanks for clarifying.

**Comment by funilrys** [comment]
Hi @StevenBlack,
Funceble test if a domain or a list of domain is/are still active!

Here's a short summary:

* **ACTIVE**
    * Date of expiration accessible per WHOIS-Server (`whois` command)
    * `nslookup` don't return "**server can't find domain-name.me: NXDOMAIN**"
* **INACTIVE**
    * `whois` server don't return anything or date is unreadable
    * `nslookup` return "**server can't find domain-name.me: NXDOMAIN**"
* **INVALID**
    * Domain extension has an invalid format or is unregistered into **[IANA](https://www.iana.org/domains/root/db) Root Zone Database**.

You simply can exclude/remove the domains under the INACTIVE directory of each list @mitchellkrogza already did for his list :smile:

**Comment by StevenBlack** [comment]
@funilrys  In your experience... I wonder: how is "inactive" prone to Type II errors –  a "false negative"?  

I'm guessing the whois test might not be great, since all whois servers practice some form of rate-limiting on requests.

**Comment by funilrys** [comment]
@StevenBlack That's why I test first whois then test `nslookup` if it doesn't return anything. I don't rest only on whois.

**To make it simple**
Whois true && nslookup true ===> active
Whois false && nslookup true ===> active
Whois true && nslookup false ===> active
whois false && nslookup false ==> inactive

But thanks for pointing out the Type II errors, you're right in a certain way...
I think I have to find a third way in order to be sure...

**Comment by FadeMind** [comment]
@funilrys @StevenBlack IMO good idea is MERGE this functionality directly in **updateHostsFile.py** code but main code is now in BASH: https://github.com/funilrys/funceble/blob/master/funceble

**Comment by funilrys** [comment]
@FadeMind It's planned to write a python version but @mitchellkrogza like to say, I'm a bash basher :smile:

**Comment by funilrys** [comment]
In betwenn @FadeMind It's not a good idea at all to merge it cause:

-  the file you're talking about gonna take too much time to execute if I implement it to it. 
    -  **updateHostsFile.py** don't read the domains one by one.

**Comment by funilrys** [comment]
Hello Steven @StevenBlack,

I may not have the experience you have as you previously said but I wanted to update the representation of the logic of funceble which is much clear now than the one I gave you in March.

I wanted a short expert opinion so can you tell me what you think about it? 

Thank you.

![](https://funilrys.com/user/pages/projects/funceble/global.png)

**Comment by StevenBlack** [comment]
Hi Nissar @funilrys 

This seems like a good start.  It appears like a reasonable approach.  But I think `whois` is the weak link in all this.  Not much can be done about that.  Maybe the flow should bypass `whois` if it times out or returns an error.

**Comment by mitchellkrogza** [comment]
@StevenBlack having very good results with funceble on this repo - https://github.com/mitchellkrogza/The-Big-List-of-Hacked-Malware-Web-Sites and run it against very large lists with over 5900 entries and it seems to work like a charm. @funilrys been doing some great work on this and I am sure over time he will perfect it even more.

**Comment by funilrys** [comment]
Thanks for the concrete answer Steven @StevenBlack. 

Funceble already has an integrated time out function written for all call of `whois` which is set by default to 30 seconds but can be changed by the user with the `-t` argument. For example, for the big list like The [Badd-Boyz-Hosts](https://github.com/mitchellkrogza/Badd-Boyz-Hosts) the time out under [dead-hosts](https://github.com/funilrys/dead-hosts/) is set to 1 second instead of 30 seconds.

For the error part, I assume that if we don't match the expiration date there is an error but I may have to review this part...

________
Thanks for the support Mitchell @mitchellkrogza.

Be sure that funceble will continue to be developed in order to be more efficient and perfect!

_________
Thanks again to both of you.

Have a nice day/night

Cheers,
Nissar
